kafka-elastic/src/main/java/com/baeldung/kafka/controller/SpringBootProducerController.java

```diff
@@ -0,0 +1,33 @@
+package com.baeldung.kafka.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import com.baeldung.kafka.model.NotificationModel;
+import com.baeldung.kafka.service.KafkaProducerService;
+
+@RestController
+@RequestMapping("/api/v1")
+public class SpringBootProducerController {
+
+    @Autowired
+    KafkaProducerService kafkaProducerService;
+
+    @GetMapping("/status")
+    public String checkMethod() {
+        return "working";
+    }
+
+    @PostMapping("/send")
+    public ResponseEntity<?> send(@RequestBody NotificationModel notificationModel) {
+        try {
+            kafkaProducerService.sendMessage(notificationModel);
+        } catch (Exception e) {
+            return ResponseEntity.internalServerError()
```

Comment:
We lose the stack trace here

Reply:
That's the last layer. Where else can we propagate it beyond the controller ? e.getMessage() is already added to message body.

Reply:
The usual pattern is to log the error with the stack trace preserved, to make it feasible to investigate the cause when investigating the error message

Reply:
incoprated